OpenRA.Mods.Common/Traits/Buildings/Refinery.cs

```diff
@@ -75,10 +75,11 @@ public class Refinery : INotifyCreated, ITick, IAcceptResources, INotifySold, IN
 		public bool IsDragRequired => info.IsDragRequired;
 		public WVec DragOffset => info.DragOffset;
 		public int DragLength => info.DragLength;
+		public bool IsAliveAndInWorld => Self.IsInWorld && !Self.IsDead;
```

Comment:
I'm not sure wether I agree with the current convention. What's the problem with traits having their actor reference public?

Reply:
If anything it should be called `Actor` not `Self`. A convention is a convention and not about the opposite being technically unfeasible.

Reply:
But this convention makes it so easy to make mistakes by using / passing the wrong reference to an actor or a trait

Reply:
> IMO it would be better to keep the activities using Actor refinery but also pass in a RefineryInfo refineryInfo. This would allow all of the otherwise unnecessary properties just above to be removed.

I will be passing in Dock trait. and It really feels wrong to pass both the Dock trait and the Dock actor separately. This duplication just feels wrong

Reply:
This has caused me so many headackes when working with various parts of the engine. I am really unsure why we have such a convention